### DISCUSSION TITLE
Blocks powered by Hypermedia controls

### DISCUSSION THREAD

**Discussion by EstebanForge** [TCattd/HTMX-API-WP#6] [Ideas]
The ability to create blocks for Gutenberg, powered by Hypermedia (HTMX, Datastar, Alpine Ajax).

Could this be useful?

If so, how to implement blocks into this plugin?

Ideally just with pure PHP for the end user (devs), similar to how ACF or Carbon Fields does it.

**Comment by jasalt** [comment]
@TCattd Just found your plugin which seems like a more polished version of Htmxpress that has example of this functionality here https://github.com/svandragt/htmxpress-serverside-block/. Maybe it could give some ideas.

**Reply by EstebanForge** [reply]
Thanks for the kind words.

**Comment by jasalt** [comment]
Also, personally I have been moving from ACF to Carbon Fields which is code oriented and does not include paywalls etc. Interested to see how it could be compatible also. Block example https://docs.carbonfields.net/learn/containers/gutenberg-blocks.html.

**Reply by EstebanForge** [reply]
I mentioned ACF, but because i like that they offer the option to create blocks with pure PHP.
But yeah, having something more... universal, not paid, would be awesome here.

I have also used Carbon Fields before. It's nice indeed. Have it on sight for this.

Pods https://pods.io is another one that can do Blocks, if i'm not mistaken.

But i haven't decided yet what to use to help me here. I definitely don't want to spend time creating such a library myself (for fields and store-read meta data). Not now.

Esteban

**Comment by EstebanForge** [comment]
And done: https://github.com/EstebanForge/HyperPress/blob/main/CHANGELOG.md#300--2025-08-21

Available 3.0.0

**Reply by jasalt** [reply]
Very interesting. I've been experimenting with Datastar and it's great.

After dropping "block builder" tools and experimenting with the vanilla block API's on a couple FSE theme projects I ended up taking different approach writing Twig templates in the editor which get connected to "view controller logic" via Block Bindings API (["twig-templating-block"](https://github.com/jasalt/twig-templating-block)). While it does not take anything AJAX related into account there seems to be some overlap.

Also, went with SCF (ACF fork) just for the meta fields as the paywall got removed, but will look how this "HyperFields" is set up and what other ideas are there.
